### PR TITLE
iPhone Keyboard Layout

### PR DESCRIPTION
**Pull request by jsmorgan42**
Changes included:
- Updated layouts for iPhone for the keyboard for landscape & portrait
- Dynamically changes keyboard from QWERTY to alphabetical when changing orientation form landscape to portrait
- Updated top bar to show save/unsave button only in keyboard mode

Screen Shot 2020-03-05 at 2 27 24 PM
Screen Shot 2020-03-05 at 2 27 35 PM

### CODE REVIEW

File: EyeSpeak/Features/Keyboard/KeyboardKeyCollectionViewCell.swift [hunk 1]
```diff
@@ -33,21 +28,27 @@ class KeyboardKeyCollectionViewCell: VocableCollectionViewCell {
...
     func setup(with image: UIImage?) {
-        guard let image = image else {
+        guard let image = image, let font = textLabel.font else {
             return
         }
         
+        let size = CGSize(width: bounds.width, height: font.ascender)
+        
+        let scaledCellBounds = CGRect(origin: .zero, size: size)
+        let scaledRect = AVMakeRect(aspectRatio: image.size, insideRect: scaledCellBounds)
```
Comment (pg8wood) [inline EyeSpeak/Features/Keyboard/KeyboardKeyCollectionViewCell.swift:45]: Dang that's neat

File: EyeSpeak/Features/Keyboard/KeyboardKeyCollectionViewCell.xib [hunk 2]
```diff
@@ -17,9 +17,15 @@
...
                     <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Label" textAlignment="center" lineBreakMode="tailTruncation" numberOfLines="0" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="57z-XF-Jde">
                         <rect key="frame" x="16" y="0.0" width="237" height="218"/>
-                        <fontDescription key="fontDescription" style="UICTFontTextStyleTitle1"/>
+                        <fontDescription key="fontDescription" type="boldSystem" pointSize="48"/>
                         <nil key="textColor"/>
                         <nil key="highlightedColor"/>
+                        <variation key="widthClass=compact">
+                            <fontDescription key="fontDescription" type="boldSystem" pointSize="34"/>
+                        </variation>
+                        <variation key="heightClass=compact-widthClass=regular">
```
Comment (pg8wood) [inline EyeSpeak/Features/Keyboard/KeyboardKeyCollectionViewCell.xib]: Do we want the the smaller font for compact width as well?

File: EyeSpeak/Features/Keyboard/SuggestionCollectionViewCell.swift [hunk 3]
```diff
@@ -8,13 +8,44 @@
...
-            super.setup(title: title)
+            textLabel.text = title
+        } else {
+            textLabel.text = "\"" + title + "\""
+        }
+    }
+    
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        adjustBackgroundColorForSizeClass()
+    }
```
Comment (pg8wood) [inline EyeSpeak/Features/Keyboard/SuggestionCollectionViewCell.swift]:
```suggestion
    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
        super.traitCollectionDidChange(previousTraitCollection)
        adjustBackgroundColorForSizeClass()
    }
```

File: EyeSpeak/Features/Presets/Models/Keyboard.swift [hunk 4]
```diff
@@ -0,0 +1,14 @@
+//
+//  Keyboard.swift
```
Comment (pg8wood) [inline EyeSpeak/Features/Presets/Models/Keyboard.swift]: Thoughts on making the filename match the struct for easy searching?

**Review by Clstroud — APPROVED** [review]
Looks good to me, but I helped so I'm a little biased. Should probably get other eyes on it.

File: EyeSpeak/Features/Presets/PresetsViewController.swift [hunk 5]
```diff
@@ -92,11 +92,7 @@ class PresetsViewController: UICollectionViewController {
         }
     }
     
-    private var showKeyboard: Bool = false {
-        didSet {
```
Comment (pg8wood) [inline EyeSpeak/Features/Presets/PresetsViewController.swift:96]: 👋

**Review by pg8wood — APPROVED** [review]
👍

File: EyeSpeak/Features/Presets/Models/Keyboard.swift (hunk 4 above)
Comment (jsmorgan42) [reply EyeSpeak/Features/Presets/Models/Keyboard.swift]: I guess the intention was that there would be a class called keyboard could include various data model structs related to the keyboard.

File: EyeSpeak/Features/Keyboard/KeyboardKeyCollectionViewCell.xib (hunk 2 above)
Comment (jsmorgan42) [reply EyeSpeak/Features/Keyboard/KeyboardKeyCollectionViewCell.xib]: Yep, nice catch 💯

File: EyeSpeak/Features/Presets/PresetsViewController.swift [hunk 6]
```diff
@@ -18,7 +18,7 @@ class PresetsViewController: UICollectionViewController {
     
     private enum HintText: String, CaseIterable {
         case preset = "Select something below to speak"
-        case keyboard = "Select letters below to start typing."
+        case keyboard = "Start typing..."
```
Comment (stevefosterwta) [inline EyeSpeak/Features/Presets/PresetsViewController.swift]: Should these be localized?

Comment (jsmorgan42) [reply EyeSpeak/Features/Presets/PresetsViewController.swift]: I can move these into the newly created Keyboard class that contains the data models 👍

**Pull request merged**